### PR TITLE
Update some link labels on My VA v2

### LINKED ISSUE #23776: [FE] My VA: "Learn how to file a VA disability claim" link text is incorrect on staging
## Background
On My VA dashboard on staging, in the Apply for VA benefits section, under the Disability compensation subsection, there is a link that reads "Learn how to file a claim for disability." This copy is incorrect. It should read "Learn how to file a VA disability claim"

## Steps to reproduce
1. Log in to staging with user 10
2. Scroll to Apply for VA benefits section
3. Locate the Disability compensation subsection
4. See that the link at the bottom of the small paragraph reads "Learn how to file a claim for disability"
5. Compare to text as indicated in [documentation](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity-personalization/my-va/2.0-redesign/frontend/documentation/my-va-copy.md) and [invision](https://vsateams.invisionapp.com/console/share/SH10HT8JCKYM/600538559)

## Screenshot
Staging:
image

### LINKED ISSUE #23779: [FE] My VA: "Learn how to apply for education benefits" link text is incorrect on staging
## Background
On My VA dashboard on staging, in the Apply for VA benefits section, under the Education and training subsection, there is a link that reads "Learn how to apply for education benefits." This copy is incorrect. It should read "Learn how to apply for VA education benefits."

## Steps to reproduce
1. Log in to staging with user 10
2. Scroll to Apply for VA benefits section
3. Locate the Education and training subsection
4. See that the link at the bottom of the small paragraph reads "Learn how to apply for education benefits" when it should say "Learn how to apply for VA education benefits."
5. Compare to text as indicated in [documentation](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity-personalization/my-va/2.0-redesign/frontend/documentation/my-va-copy.md) and [invision](https://vsateams.invisionapp.com/console/share/SH10HT8JCKYM/600538559)

## Screenshot
Staging:
image

## Tasks
- [x] Correct link text to read "Learn how to apply for VA education benefits" instead of "Learn how to apply for education benefits"

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
Updates two link labels in the My VA "Apply for benefits" section

Closes department-of-veterans-affairs/va.gov-team#23776 and closes department-of-veterans-affairs/va.gov-team#23779

## Testing done
Unit tests, which I refactored heavily.

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.unit.spec.jsx [hunk 1]
```diff
@@ -32,6 +32,69 @@ function oneYearFromNow() {
   return Date.now() + oneYearInMS;
 }
 
+function loadingSpinnerIsHidden(view) {
```
Comment (erikphansen) [inline src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.unit.spec.jsx:35]: I may have gone a little overboard with the test refactoring. I was trying to do two things:

1. Stop throwing around the same strings/regexes in multiple places
2. But mostly make the tests themselves easier to read

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.unit.spec.jsx [hunk 2]
```diff
@@ -299,20 +361,11 @@ describe('ApplyForBenefits component', () => {
...
-            name: /learn how to file a claim/i,
-          });
-          view.getByRole('link', {
-            name: /learn how to apply for education benefits/i,
-          });
+          loadingSpinnerIsHidden(view);
+
+          healthCareInfoIsShown(view);
+          claimsInfoIsShown(view);
+          educationInfoIsShown(view);
```
Comment (erikphansen) [inline src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.unit.spec.jsx:368]: ☝️ this is what I was really after. I think this much nicer to read now.

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.unit.spec.jsx [hunk 3]
```diff
@@ -396,21 +444,9 @@ describe('ApplyForBenefits component', () => {
...
-            }),
-          ).to.exist;
-          expect(
-            view.getByRole('link', {
-              name: /learn how to apply for education benefits/i,
-            }),
-          ).to.exist;
+          healthCareInfoIsShown(view);
+          claimsInfoIsShown(view);
+          educationInfoIsShown(view);
```
Comment (erikphansen) [inline src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.unit.spec.jsx:449]: So much nicer. Plus, now we could make these helper function more robust (check for more elements to be visible or hidden) very easily.

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.jsx [hunk 4]
```diff
@@ -143,7 +147,7 @@ const ApplyForBenefits = ({
           <BenefitOfInterest
             title="Disability compensation"
             icon="disability"
-            ctaButtonLabel="Learn how to file a claim for disability"
+            ctaButtonLabel="Learn how to file a VA disability claim"
```
Comment (erikphansen) [inline src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.jsx:150]: the real purpose of this PR

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.jsx [hunk 5]
```diff
@@ -156,7 +160,7 @@ const ApplyForBenefits = ({
             <BenefitOfInterest
               title="Education and training"
               icon="education"
-              ctaButtonLabel="Learn how to apply for education benefits"
+              ctaButtonLabel="Learn how to apply for VA education benefits"
```
Comment (erikphansen) [inline src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.jsx:163]: the real purpose of this PR

**Review by micahchiang — APPROVED** [review]

**Pull request merged**